base/sparse/cholmod.jl

```diff
@@ -1,75 +1,62 @@
 module CHOLMOD
 
-export                                  # types
- CholmodDense,
- CholmodFactor,
- CholmodSparse,
- CholmodTriplet,
- CholmodDense!,
- etree
-
-import Base: (*), convert, copy, ctranspose, eltype, findnz, getindex, hcat,
-             isvalid, show, size, sort!, transpose, vcat
+import Base: (*), convert, copy, eltype, getindex, show, size
 
 import Base.LinAlg: (\), A_mul_Bc, A_mul_Bt, Ac_ldiv_B, Ac_mul_B, At_ldiv_B, At_mul_B,
-                 A_ldiv_B!, cholfact, cholfact!, copy, det, diag,
-                 full, isposdef!, logdet, norm, scale, scale!
+                 A_ldiv_B!, cholfact, cholfact!, det, diag, full, ishermitian, isposdef,
+                 issym, isvalid, ldltfact, logdet
+
+import Base.SparseMatrix: sparse
+
+export
+    Dense,
+    Factor,
+    Sparse
+
+using Base.SparseMatrix: AbstractSparseMatrix, SparseMatrixCSC, increment, increment!, indtype, decrement, decrement!
 
-importall Base.SparseMatrix
-import Base.SparseMatrix: increment, increment!, decrement, decrement!
+#########
+# Setup #
+#########
 
 include("cholmod_h.jl")
 
 macro isok(A)
-    :($A==CHOLMOD_TRUE || throw(CholmodException()))
+    :($A == TRUE || throw(CHOLMODException("")))
 end
 
-const chm_ver    = Array(Cint, 3)
+const version_array = Array(Cint, 3)
 if dlsym(dlopen("libcholmod"), :cholmod_version) != C_NULL
-    ccall((:cholmod_version, :libcholmod), Cint, (Ptr{Cint},), chm_ver)
+    ccall((:cholmod_version, :libcholmod), Cint, (Ptr{Cint},), version_array)
 else
-    ccall((:jl_cholmod_version, :libsuitesparse_wrapper), Cint, (Ptr{Cint},), chm_ver)
-end
-const chm_com_sz = ccall((:jl_cholmod_common_size,:libsuitesparse_wrapper),Int,())
-const chm_com    = fill(0xff, chm_com_sz)
-const chm_l_com  = fill(0xff, chm_com_sz)
-## chm_com and chm_l_com must be initialized at runtime because they contain pointers
-## to functions in libc.so, whose addresses can change
-function cmn(::Type{Int32})
-    if isnan(reinterpret(Float64,chm_com[1:8])[1])
-        @isok ccall((:cholmod_start, :libcholmod), Cint, (Ptr{UInt8},), chm_com)
-    end
-    chm_com
-end
-function cmn(::Type{Int64})
-    if isnan(reinterpret(Float64,chm_l_com[1:8])[1])
-        @isok ccall((:cholmod_l_start, :libcholmod), Cint, (Ptr{UInt8},), chm_l_com)
-    end
-    chm_l_com
+    ccall((:jl_cholmod_version, :libsuitesparse_wrapper), Cint, (Ptr{Cint},), version_array)
```

Comment:
I think if `dlsym` does not find the `cholmod_version` symbol, it is safe to assume that the version of CHOLMOD is too old for us. This lets us get rid of `jl_cholmod_version` from the wrapper too.